### PR TITLE
Zonotope membership test using simplex

### LINKED ISSUE #246: Counterexample for containment check in zonotope
Let:

```julia
c = [0, 0.]
G =  [0.75  -0.05  1.0  1.0   0.25  0.05   0.0;
      0.5    0.95  2.5  1.0  -0.5   0.05  -1.5]
Z = Zonotope(c, G)
```

the result of `[3., 3] ∈ Z` is false but should be `true`.

### PR DESCRIPTION
**Pull request by mforets**
Closes #246.

### CODE REVIEW

**Comment by mforets** [comment]
> I do not understand what happens here. I see that you add one dimension for the objective function, but why does the solution have to be optimal? Can you add some more documentation what happens here?

i improved the docstring (and removed the previous one, i didn't see that it was above!)

for the type-parametric issue, i would have to investigate it further how we can enable this, since the computations ultimately relies on the LP solver backend used.

**Comment by schillic** [comment]
Thanks. I suggest to just test what happens when you input a `Zonotope{Rational{Int}}`. If yes, then we can leave it as is. Otherwise we should at least crash with a clear explanation.

File: src/Zonotope.jl [hunk 1]
```diff
@@ -207,36 +207,28 @@ false
...
+    p, n = ngens(Z), dim(Z)
+    A = [[1.; fill(0, p)]'; [fill(0., n) Z.generators]]
+    b = [0.; (x - Z.center)]
+    lbounds = [0; fill(-1., p)]
+    ubounds = [Inf; fill(1., p)]
+    sense = ['>'; fill('=', n)]
+    obj = [1.; fill(0, p)]
+
+    lp = linprog(obj, A, sense, b, lbounds, ubounds, solver)
+    res = (lp.status == :Optimal) # Infeasible of Unboudned => false
```
[the remark this answers is not in the document]
Comment (schillic) [reply src/Zonotope.jl]: So it should be "if" instead of "of"?

[the remark this answers is not in the document]
Comment (mforets) [reply src/Zonotope.jl]: yes, thanks

File: src/Zonotope.jl [hunk 2]
```diff
@@ -173,30 +173,22 @@ function σ(d::AbstractVector{<:Real}, Z::Zonotope)::AbstractVector{<:Real}
...
 
 Check whether a given point is contained in a zonotope.
 
 ### Input
 
-- `x` -- point/vector
-- `Z` -- zonotope
+- `x`      -- point/vector
+- `Z`      -- zonotope
+- `solver` -- (optiona, default: `GLPKSolverLP(method=:Simplex)`) the backend
```
Comment (schillic) [inline src/Zonotope.jl]: "optional"

**Comment by schillic** [comment]
Is there a reason why you use `fill` instead of `zeros`/`ones`?

**Comment by mforets** [comment]
it works using `method=:Exact`; compare:

```julia
julia>     lp = linprog(obj, A, sense, b, lbounds, ubounds, GLPKSolverLP(method=:Exact))
glp_exact: 3 rows, 3 columns, 3 non-zeros
GNU MP bignum library is being used
      0:   infsum =                    0.2   (0)
      2:   infsum =                      0   (0)
*     2:   objval =                      0   (0)
*     2:   objval =                      0   (0)
OPTIMAL SOLUTION FOUND
MathProgBase.HighLevelInterface.LinprogSolution(:Optimal, 0.0, [0.0, 0.0, 0.0], Dict{Any,Any}(Pair{Any,Any}(:lambda, [0.0, 0.0, 0.0]),Pair{Any,Any}(:redcost, [1.0, 0.0, 0.0])))
```
and

```julia
julia>     lp = linprog(obj, A, sense, b, lbounds, ubounds, GLPKSolverLP(method=:Simplex))
MathProgBase.HighLevelInterface.LinprogSolution(:Optimal, 0.0, [0.0, 0.0, 0.0], Dict{Any,Any}(Pair{Any,Any}(:lambda, [0.0, 0.0, 0.0]),Pair{Any,Any}(:redcost, [1.0, 0.0, 0.0])))
```

i wouldn't expect that for the former case it returns values in `Float64` though. but clearly  `glp_exact` is triggered.

**Comment by mforets** [comment]
> Is there a reason why you use fill instead of zeros/ones?

no

**Comment by schillic** [comment]
>  i wouldn't expect that for the former case it returns values in Float64 though. but clearly glp_exact is triggered.

Yes, that looks strange, but maybe it can automatically work with floats if possible for efficiency.

**Pull request merged**